sphinx/domains/python.py

```diff
@@ -614,23 +615,37 @@ def get_index_text(self, modname: str, name_cls: Tuple[str, str]) -> str:
             return ''
 
 
-class PyMethod(PyObject):
+class PyClassMemberBase(PyObject, ABC):
+    def _split_member_name(self, mod: str, name: str) -> Tuple[Optional[str], str]:
+        try:
+            clsname, selfname = name.rsplit('.', 1)
+            if mod and self.env.config.add_module_names:
+                clsname = '.'.join([mod, clsname])
+            return clsname, selfname
+        except ValueError:
+            if mod:
+                return None, _('%s() (in module %s)') % (name, mod)
+            elif self.needs_arglist():
+                return None, '%s()' % name
+            else:
+                return None, name
+
+
+class PyMethod(PyClassMemberBase):
     """Description of a method."""
 
     option_spec = PyObject.option_spec.copy()
```

Comment:
I don't know why mypy warns this line. But it seems `PyObject.option_spec` is needed to annotated like `ObjectDescription.option_spec`:
https://github.com/sphinx-doc/sphinx/blob/372813a137cd40a13a3be55cd64cbb48c08c41c2/sphinx/directives/__init__.py#L63-L65

Could you try to add it please?

Reply:
OK, makes sense!